### DISCUSSION TITLE
Release management

### DISCUSSION THREAD

**Discussion by sjib** [QGEP/QGEP#732] [Ideas]
We currently have quite a lot of different releases:

1. QGEP datamodel https://github.com/QGEP/datamodel/releases/ - currently 1.5.4
2. QGEP Plugin https://github.com/QGEP/qgepplugin/releases/latest - currently 1.5.5
3. QGEP Project file https://github.com/QGEP/QGEP/releases/latest - currently v8.0
4. QGIS qgepqwat2ili https://github.com/QGEP/qgepqwat2ili/tags - currently v7.0
5. QGEP SWMM (integrated in the plugin and the datamodel) - no release version
6. QGIS versions https://www.qgis.org/de/site/forusers/download.html - currently 3.24.1 and 3.22.5 as LTR
7. In addition we use a Java library that has to fit ili2pg in ModelBaker
8. and finally we use Postgres as database that has to fit with Java so that our Datamodel tool works fine

So the proposal came up to inforce a stronger **Release management** that maybe also harmonizes the release numbers, at least where QGEP can control it.

On the other hand we have a lot of ideas and open issues and we should again set the priorities how we set the priorities for implementation.

QGIS uses Milestone to plan their open issues: https://github.com/qgis/QGIS/milestones a concept I think we could also use and maintain by the Technical group and within that group one (or two) people that take extra time to pre-sort the priorities.

There is also a new feature call Projects in github: https://github.com/orgs/QGEP/projects?type=beta that we could use and define a project per new release.

At the end we should get an easy overview 

Release - QGIS Version - Java library - Postgres version
1.6        - 3.24                - 11.x            - 12

Each time when one of the release 1 - 5 changes the new QGEP release number should go x.1 up

What do you think of this ideas? Your thoughts on this topic are welcomed.

**Comment by ponceta** [comment]
To consider in the discussion

For the drinking water module (QWAT) we have:

1. QWAT datamodel https://github.com/qwat/qwat-data-model/releases - currently 1.3.6
2. QWAT QGIS project files https://github.com/qwat/QWAT/releases - no release yet

Extensions
3. QWAT SIRE extension
4. QWAT aquafri extension
5. QWAT FR raepa extension

Plugins
1. QGIS qgepqwat2ili plugin https://github.com/QGEP/qgepqwat2ili/tags - currently v7.0 (shared with QGEP)
2. QGIS PG history viewer plugin (shared with QGEP) (future - to be shared with all TEKSI modules)
3. QGIS Landsurvey code import plugin (future - to be shared with all TEKSI modules)
4. QGIS ModelBaker plugin ()

**Comment by sjib** [comment]
3 level of releases:

- **bugfix release** x.x.1: whenever a bugfix is ready
- **minor change** x.1.1: minor feature release, backward compatible including documentation
- **major release** 1.1.1: major feature release, not backward compatible

Triggers to create a release: bug fix, adapting a feature, new feature, new QGIS release that needs adaption, datamodel change, ...

Minor and major release should be schedulded and be known ahead of time.

**Comment by sjib** [comment]
Start of a TEKSI Roadmap: https://github.com/teksi/Home/wiki/TEKSI-modules-releases-and-roadmap
